### ISSUE TITLE
[Feature request] Installing daemon for specified service only

### ISSUE THREAD

**Issue by Vaults** [newsnowlabs/docker-ingress-routing-daemon#1]
As a user with a very diverse swarm (30 different services), testing and running the routing daemon would be very convenient if only active on a certain service with certain ports, for instance :80 and :443 only.

I imagine it would be feasible with the following changes (I've thought ahead to hopefully relieve you of work if the concept is solid):

1. Adding a service name variable, e.g. ```DAEMON_SERVICE```
2. Getting the exposed ports with 
```bash 
PORTS=docker $DAEMON_SERVICE inspect nginx -f '{{range .Spec.EndpointSpec.Ports}}{{.PublishedPort}} {{end}}'
```
3. For each port, add SNAT skip rules (if single service):
```bash
for PORT in $PORTS
do
    # Add a rule ahead of the ingress network SNAT rule, that will cause the SNAT rule to be skipped.
    log "Adding ingress_sbox iptables nat rule: iptables -t nat -I POSTROUTING -d $INGRESS_SUBNET -m ipvs --ipvs -j ACCEPT"
    nsenter --net=/var/run/docker/netns/ingress_sbox iptables -t nat -I POSTROUTING -d $INGRESS_SUBNET -dport $PORT -m ipvs --ipvs -j ACCEPT

    # 1. Set TOS to NODE_ID in all outgoing packets to INGRESS_SUBNET
    log "Adding ingress_sbox iptables mangle rule: iptables -t mangle -A POSTROUTING -d $INGRESS_SUBNET -dport $PORT -j TOS --set-tos $NODE_ID/0xff"
    nsenter --net=/var/run/docker/netns/ingress_sbox iptables -t mangle -A POSTROUTING -d $INGRESS_SUBNET -dport $PORT -j TOS --set-tos $NODE_ID/0xff

    log "Disabling ingress_sbox connection tracking"
    nsenter --net=/var/run/docker/netns/ingress_sbox iptables -t raw -I PREROUTING -p tcp -j CT --notrack
done
```

4. The same needs to be done for removal. I think automatic cleaning up like the script already has will be non-trivial then.

5. Changing the service conditional 
```bash
#old
if [ -n "$SERVICE" ]; then
#new
if [ -n "$DAEMON_SERVICE"  -a "$SERVICE" -eq "$DAEMON_SERVICE" ] || [ !-n "$DAEMON_SERVICE"]; then
```

If I'm not mistaken, these daemon changes would also allow for running it multiple times (so you can configure as many services as you'd like). Thank you for reading, I hope you have the time to evaluate this :) You may reuse any part mentioned above.

**Comment by struanb** [comment]
Hi @Vaults. Thanks for thinking of and proposing this. It would be useful to have the option for firewall rules to only match the necessary ports, so it could coexist with the legacy SNAT-based behaviour.

Few things I'm thinking:
- if we can use `-m multiport --dports <comma-separated-ports>` it will be more efficient than multiple iptables rules
- not sure it's a problem to set the TOS byte irrespective of whether the dports match; I guess only doing so would be 'safer' so we might as well do it, but in practice the TOS byte will most likely be ignored by all other parts of your network
- it could be useful to retain the current behaviour as an option (where the routing rules will be applied to all services, rather than just named services) for users who want that, so that they don't have to restart the daemon before launching a new service.

I'll try and patch this all together and test it in the coming few days.

**Comment by struanb** [comment]
An implementation of this, along with some other improvements, is nearly ready. The new version allows you to specify the TCP and/or UDP ports you wish to disable masquerading for.

I need to update documentation, then will release as v3.0.0.

**Comment by Vaults** [comment]
@struanb Amazing! Thank you so much :)

**Comment by struanb** [comment]
The new code and documentation (README) is released as v3.0.0. Please check it out and raise new issues for any regressions or new problems.

I'll close this now.
